core/build.gradle.kts

```diff
@@ -148,6 +149,7 @@ kotlin {
         commonMain {
             dependencies {
                 api("org.jetbrains.kotlin:kotlin-stdlib-common")
+                compileOnly("org.jetbrains.kotlinx:kotlinx-serialization-core:1.0.1")
```

Comment:
How thoroughly? I checked that the tests do pass, but is there potential for more subtle breakage?

Reply:
I mean, why we should release datetime library with an obsolete dependency.

Also, AFAIR we investigated `compileOnly` dependency and found out that it may lead to problems, didn't we?

Reply:
It did cause breakage with autogenerated serializers, as they add new constructors, which caused the classes to throw on load. Rewriting everything to use autogenerated serializers fixed the crashes.